File: SOSAuto_BACKEND/src/main/java/com/sosauto_backend/model/Dto/DemandeDepannageDTO.java
package com.sosauto_backend.model.Dto;

import com.sosauto_backend.model.Entity.Automobiliste;
import com.sosauto_backend.model.Entity.Mécanicien;
import com.sosauto_backend.model.Entity.Panne;
import com.sosauto_backend.model.Enum.EtatPanne;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class DemandeDepannageDTO {

    private Long demandeid;
    private LocalDateTime dateTime;
    private Double latitude;
    private Double longitude;
    private String description;
    private EtatPanne etat;
    private Automobiliste automobiliste;
    private Mécanicien mécanicien;
    private Panne panne;
}

File: SOSAuto_FRONTEND/src/app/services/models/demande-depannage-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AutomobilisteDto } from '../models/automobiliste-dto';
import { Mecanicien } from '../models/mecanicien';
import { Panne } from '../models/panne';
export interface DemandeDepannageDto {
  automobiliste?: AutomobilisteDto;
  dateTime?: string;
  demandeid?: number;
  description?: string;
  etat?: 'A_FAIRE' | 'EN_COURS' | 'TERMINE';
  latitude?: number;
  longitude?: number;
  'mécanicien'?: Mecanicien;
  panne?: Panne;
}
